pkg/asset/machines/baremetal/hosts.go

```diff
@@ -29,8 +31,14 @@ type HostSettings struct {
 	NetworkConfigSecrets []corev1.Secret
 }
 
-func createNetworkConfigSecret(host *baremetal.Host) corev1.Secret {
-	return corev1.Secret{
+func createNetworkConfigSecret(host *baremetal.Host) (*corev1.Secret, error) {
+
+	yamlNetworkConfig, err := yaml.Marshal(host.NetworkConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error while creating network config secret")
```

Comment:
Maybe we can note that this field only can be in yaml format?

Reply:
Doc updated